api/views.py

```diff
@@ -0,0 +1,23 @@
+from rest_framework import generics
+from .models import address
+from .serializers import AddressGetSerializer, AddressCreateSerializer
+
+class AddressListView(generics.ListAPIView):
+    # read-only access to a list of addresses.
+    queryset = address.objects.all()
+    # serializer for rendering the output.
+    serializer_class = AddressGetSerializer
+
+
+class AddressRetrieveView(generics.RetrieveAPIView):
+    # read-only access to a single address instance.
+    queryset = address.objects.all()
```

Comment:
how is this `single`? both this and above are using `all()`.

Reply:
Internally, DRF handles retrieving a single instance from the queryset based on the provided lookup value (usually the primary key) when we use a generic view like RetrieveAPIView. The all() method doesn't necessarily retrieve all instances from the database. Instead, it provides a queryset that DRF can use to efficiently retrieve the specific instance you're interested in based on the lookup value provided in the URL.